docarray/array/array/io.py

```diff
@@ -302,6 +304,56 @@ def to_json(self) -> str:
         """
         return json.dumps([doc.json() for doc in self])
 
+    @classmethod
+    def _check_for_valid_document_type(cls) -> None:
+        if cls.document_type == AnyDocument:
+            raise TypeError(
+                'There is no document schema defined. '
+                'Please specify the DocumentArray\'s Document type using `DocumentArray[MyDoc]`.'
+            )
+
+    @classmethod
+    def _check_for_valid_access_paths(cls, field_names: Optional[List[str]]) -> None:
+        if field_names is None or len(field_names) == 0:
+            raise TypeError("No field names are given.")
+
+        valid = [
+            is_access_path_valid(cls.document_type, field) for field in field_names
+        ]
+        if not all(valid):
+            raise ValueError(
+                f'Column names do not match the schema of the DocumentArray\'s document type '
+                f'({cls.document_type.__name__}): {list(compress(field_names, [not v for v in valid]))}'
+            )
+
+    @staticmethod
+    def access_path_dict_to_nested_dict(
```

Comment:
didn't you already implement something like this in your previous PR? or am i mixing things up?

Reply:
ok sounds good, yes keeping them in the same helper file would be ideal